### ISSUE TITLE
Generate URI/JSON for a connection from the Airflow Connection UI

### ISSUE THREAD

**Issue by manmeetkaur** [apache/airflow#29656]
### Description

Ability to generate the Connection String in either URI or JSON Format from the Connection UX on the Airflow UI.

### Use case/motivation

Many a times, have seen Data Engineers struggle with create Connection via the UI, but then struggle to generate the corresponding JSON or URI required to port it to Secret Managers. We have to either go via Astro CLI or Airflow Shell to export the connection in JSON or URI format. 

The goal is to make it easier for Users to do this via the UI, and make the migration from a Dev Environment to a more organized Prod Environment's Secret Manager much easier by reducing the turnaround time.

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
AFAIK it is not possible retrieve `password` field in Webserver/REST API, potentially for security reason

**Comment by potiuk** [comment]
correct. Passwords in the UI are "write only" for that very reason. We are not supposed to read them via UI. And we also have all the - easy - ways to get the URL either via a small snippets of Python code or simply using hte `airflow cli`  - https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#uri-format - even from exsting connections

**Comment by potiuk** [comment]
closing as this is not very likely to be implemented.

**Comment by ashb** [comment]
An option: have a button on the New Connection form that gives "Copy as URI".

**Comment by fritz-astronomer** [comment]
There is an API route to get the connection - and as mentioned a CLI route. 

I think this request is reasonable to expose the same functionality in the UI - even potentially with a password/sensitive info redacted or only in the connection creation modal, as the newest users are likely to use that.

**Comment by potiuk** [comment]
Sure. If you think so. Without sensitive info it might be fine - the 'get as uri' is kinda cumbersome though :)
